modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java

```diff
@@ -0,0 +1,162 @@
+import io.fluo.core.util.UtilWaitThread;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Run end to end test with lots of collisions and verify the following :
+ *
+ * <p>
+ * <ul>
+ * <li/>LoaderExecutor works correctly w/ lots of collisions
+ * <li/>Observers work correctly w/ lots of collisions
+ * <li/>Fluo GC works correctly after lots of collisions
+ * </ul>
+ *
+ */
+public class CollisionIT extends ITBaseMini {
+  static TypeLayer typeLayer = new TypeLayer(new StringEncoder());
+
+  static class NumLoader extends TypedLoader {
+
+    int num;
+
+    NumLoader(int num) {
+      this.num = num;
+    }
+
+    @Override
+    public void load(TypedTransactionBase tx, Context context) throws Exception {
+      tx.mutate().row(num).fam("stat").qual("total").increment(1);
+      tx.mutate().row(num).fam("stat").qual("changed").weaklyNotify();
+    }
+  }
+
+  public static class TotalObserver extends TypedObserver {
+
+    @Override
+    public ObservedColumn getObservedColumn() {
+      return new ObservedColumn(typeLayer.bc().fam("stat").qual("changed").vis(), NotificationType.WEAK);
+    }
+
+    @Override
+    public void process(TypedTransactionBase tx, Bytes row, Column col) {
+      int total = tx.get().row(row).fam("stat").qual("total").toInteger();
+      int processed = tx.get().row(row).fam("stat").qual("processed").toInteger(0);
+
+      tx.mutate().row(row).fam("stat").qual("processed").set(total);
+      tx.mutate().row("all").fam("stat").qual("total").increment(total - processed);
+    }
+  }
+
+  @Override
+  protected List<ObserverConfiguration> getObservers() {
+    return Arrays.asList(new ObserverConfiguration(TotalObserver.class.getName()));
+  }
+
+  @Override
+  protected void setConfig(FluoConfiguration config) {
+    config.setLoaderQueueSize(20);
+    config.setLoaderThreads(20);
+    config.setWorkerThreads(20);
+
+    // make updates in ZK related to Fluo GC more frequent
+    config.setProperty(ZookeeperUtil.ZK_UPDATE_PERIOD_PROP, "100");
+  }
+
+  @Test
+  public void testLotsOfCollisions() throws Exception {
+
+    Random rand = new Random(45734985);
+
+    int[] nums = new int[1000];
+    int[] expectedCounts = new int[5];
+
+    for (int i = 0; i < nums.length; i++) {
+      nums[i] = rand.nextInt(expectedCounts.length);
+      expectedCounts[nums[i]]++;
+    }
+
+    try (LoaderExecutor loader = client.newLoaderExecutor()) {
+      for (int num : nums) {
+        loader.execute(new NumLoader(num));
+      }
+    }
+
+    miniFluo.waitForObservers();
+
+    try (TypedSnapshot snapshot = typeLayer.wrap(client.newSnapshot())) {
+
+      for (int i = 0; i < expectedCounts.length; i++) {
+        Assert.assertEquals(expectedCounts[i], snapshot.get().row(i).fam("stat").qual("total").toInteger(-1));
```

Comment:
This is very cool test.  How about about also checking checking `processed` here?